profiles/lib/relations.lua

```diff
@@ -0,0 +1,94 @@
+-- Profile functions dealing with various aspects of relation parsing
+--
+-- You can run a selection you find useful in your profile,
+-- or do you own processing if/when required.
+
+Utils = require('lib/utils')
+
+Relations = {}
+
+-- match ref values to relations data
+function Relations.match_to_ref(relations, ref)
+
+  function calculate_scores(refs, tag_value)
+    local tag_tokens = Set(Utils.tokenize_common(tag_value))
+    local result = {}
+    for i, r in ipairs(refs) do
+      local ref_tokens = Utils.tokenize_common(r)
+      local score = 0
+
+      for _, t in ipairs(ref_tokens) do
+        if tag_tokens[t] then
+          if Utils.is_number(t) then
```

Comment:
https://github.com/Project-OSRM/osrm-backend/pull/4478#discussion_r139229061

> After rereading this code, I think the suffix case will be a nonissue. However, for [this way](http://www.openstreetmap.org/way/122502126), wouldn’t the two items in the way `ref` (`PA 88` and `PA 88 Truck`) tie because `88` appears in both items and both relations’ `ref` tags are set to `88`? We could factor in the relations’ `network` tags, sort of like what’s described in https://github.com/Project-OSRM/osrm-backend/pull/4478#discussion_r138947028.

Reply:
Maybe we can decrease points by `0.1 * N`, where N is a number of not matched tokens?
So in that case we will match to ref, where most of tokens was matched